lib/plugin.js

```diff
@@ -382,7 +420,40 @@ async function _processPosRemove (options, doc) {
   const changes = DeepDiff.getDiff({ old }, schema, options)
   const docMapped = _mappedFieldsLog({ changes, method, module, documentNumber, user })
 
+  options.connectionUri = this.model.db._connectionString
   await _saveHistoryLog(docMapped, options)
 }
 
+
+function hasMongoDuplicateProtection(self) {
+  const mongoVersion = self.model.db._connectionOptions.driverInfo.version
+  const majorVersion = mongoVersion.split(".")[0]
+
+  if(parseInt(majorVersion) >= 6) {
+    return true
+  } else {
+    return false
+  }
+}
+
+function getObject(target, dotNotation, value) {
+  const [k, ...rest] = dotNotation.split('.')
+  if (rest.length == 0) {
+    target[k] = value
+    return target
+  }
+  if (!target[k]) target[k] = {}
+  target[k] = getObject(target[k], rest.join('.'), value)
+  return target
+}
+
+function parseDotProperties(obj) {
```

Comment:
and this one `expandDotRootProperties`

Reply:
actually another thought. What about the following to keep all contained in one function:
```

function expandRootDotProperties(obj){
    const convertDotNotationToObjects = function (target, dotNotation, value) {
      const [k, ...rest] = dotNotation.split('.')
      if (rest.length == 0) {
        target[k] = value
        return target
      }
      if (!target[k]) target[k] = {}
      target[k] = convertDotNotationToObjects(target[k], rest.join('.'), value)
      return target
    }
    for (const [k, v] of Object.entries(obj)) {
      delete obj[k]
      obj = getObject(obj, k, v)
    }
    return obj
}
```